File: CCLib/src/java/lang/native/Throwable.lua
natives["java.lang.Throwable"] = natives["java.lang.Throwable"] or {}

natives["java.lang.Throwable"]["printStackTrace(Ljava/io/PrintStream;)V"] = function(this, p)
	findMethod(p, "println(Ljava/lang/String;)V")[1](asObjRef(p, "Ljava/io/PrintStream;"), asObjRef(this.name .. ": " .. (this.fields.message.value or ""), "Ljava/lang/String;"))
	findMethod(p, "println(Ljava/lang/String;)V")[1](asObjRef(p, "Ljava/io/PrintStream;"), asObjRef(this.stackTrace or "", "Ljava/lang/String;"))
end

File: CCLib/src/java/lang/Throwable.java
package java.lang;

import java.io.PrintStream;

public class Throwable {
	private String message;

	public Throwable() {
	}

	public Throwable(String message) {
		this.message = message;
	}

	public void printStackTrace() {
		printStackTrace(System.err);
	}

	public native void printStackTrace(PrintStream p);
}